From c47eeb3434f0b31eb23afffe5c5749310e743cfa Mon Sep 17 00:00:00 2001
From: root <jialamos@uc.cl>
Date: Thu, 15 Dec 2016 18:06:59 -0300
Subject: [PATCH] fixed unused variable

---
 lmic/lmic.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lmic/lmic.c b/lmic/lmic.c
index ad17a93..8231ae9 100644
--- a/lmic/lmic.c
+++ b/lmic/lmic.c
@@ -1365,6 +1365,7 @@ static bit_t processJoinAccept (void) {
     u1_t hdr  = LMIC.frame[0];
     u1_t dlen = LMIC.dataLen;
     u4_t mic  = os_rlsbf4(&LMIC.frame[dlen-4]); // safe before modified by encrypt!
+    (void) mic;
     if( (dlen != LEN_JA && dlen != LEN_JAEXT)
         || (hdr & (HDR_FTYPE|HDR_MAJOR)) != (HDR_FTYPE_JACC|HDR_MAJOR_V1) ) {
         EV(specCond, ERR, (e_.reason = EV::specCond_t::UNEXPECTED_FRAME,
-- 
1.9.1

